src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs

```diff
@@ -1224,41 +1224,75 @@ private void ImportBinaryOperation(ILOpcode opcode)
                     break;
                 case ILOpcode.mul_ovf:
                 case ILOpcode.mul_ovf_un:
-                    if (_compilation.TypeSystemContext.Target.Architecture == TargetArchitecture.ARM)
-                    {
-                        _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.LMulOfv), "_lmulovf");
-                        _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.ULMulOvf), "_ulmulovf");
-                    }
+                    _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.LMulOfv), "_lmulovf");
```

Comment:
Are these helpers actually used on all architectures?

Reply:
The condition around these helpers in the JIT is !TARGET_64BIT, so this check should match that.

https://github.com/dotnet/runtime/blob/ee501fb2c6ac901b761131b8c4760f74b5c18a62/src/coreclr/jit/morph.cpp#L8463-L8498

Reply:
I don't think so but they seemed to fail on x86 and I assume that in total they should have minimal size impact and it makes no sense to complicate the scanner with architecture checks for this single place since doing that requires checking each helper in the JIT when adding new platforms and such.

Reply:
We have the scanner so that these things can be tailored.

If you were to go with the premise that these things have minimal size, we would root all helpers all the time and delete most the code here.